starknet/src/server/stream.rs

```diff
@@ -6,22 +6,27 @@ use std::{
     task::{self, Poll},
 };
 
-use apibara_core::node::v1alpha2::{stream_server, StreamDataRequest, StreamDataResponse};
+use apibara_core::node::v1alpha2::{
+    stream_server, IngestionStatus, StatusRequest, StatusResponse, StreamDataRequest,
+    StreamDataResponse,
+};
 use apibara_node::{
     server::RequestObserver,
     stream::{new_data_stream, ResponseStream, StreamConfigurationStream, StreamError},
 };
-use futures::Stream;
-use pin_project::pin_project;
-use tonic::{metadata::MetadataMap, Request, Response, Streaming};
-use tracing_futures::Instrument;
 
+use crate::provider::{HttpProvider, Provider};
 use crate::{
     core::IngestionMessage,
     db::StorageReader,
     ingestion::IngestionStreamClient,
     stream::{DbBatchProducer, SequentialCursorProducer},
 };
+use futures::Stream;
+use pin_project::pin_project;
+use tonic::{metadata::MetadataMap, Request, Response, Status, Streaming};
+use tracing_futures::Instrument;
+use url::Url;
```

Comment:
These parameters should go before `crate::`, with a newline between the two sections.